src/components/Dashboard.tsx

```diff
@@ -0,0 +1,158 @@
+import { useUserState } from '../contexts/UserState'
+import constants from '../utils/constants'
+import strings from '../utils/constants'
+import { USER_ROLES } from '../utils/data'
+import { PermissionPolicyType, Filter } from '../utils/generated/graphql'
+import useListApplications from '../utils/hooks/useListApplications'
+import useListTemplates from '../utils/hooks/useListTemplates'
+import usePageTitle from '../utils/hooks/usePageTitle'
+import { TemplateDetails, TemplateInList } from '../utils/types'
+
+const Dashboard: React.FC = () => {
+  const {
+    userState: { templatePermissions, isNonRegistered },
+    logout,
+  } = useUserState()
+  const {
+    templatesData: { templatesByCategory },
+  } = useListTemplates(templatePermissions, false)
+  if (isNonRegistered) {
+    logout()
+    return null
+  }
+
+  usePageTitle(strings.PAGE_TITLE_HOME)
+
+  return (
+    <div id="dashboard">
+      {templatesByCategory.map(
+        ({ templates, templateCategory: { categoryIcon, categoryTitle } }) => (
+          <div key={categoryTitle} className="template-category">
+            <div className="title">
+              {categoryIcon && <Icon size="large" color="grey" name={categoryIcon} />}
+              <Header as="h4">{categoryTitle}</Header>
+            </div>
+            <div className="templates">
+              {templates.map((template) => (
+                <TemplateComponent key={template.code} template={template} />
+              ))}
+            </div>
+          </div>
+        )
+      )}
+    </div>
+  )
+}
+
+const TemplateComponent: React.FC<{ template: TemplateInList }> = ({ template }) => {
+  const { name, code, hasApplyPermission, filters } = template
+  return (
+    <div className="template">
+      <div className="title">
+        <Header as="h5">{name}</Header>
+
+        {hasApplyPermission && (
+          <Button inverted size="small" as={Link} to={`/application/new?type=${code}`} primary>
+            {constants.BUTTON_DASHBOARD_NEW}
+          </Button>
+        )}
+      </div>
+
+      <div className="filters">
+        {filters.map((filter) => (
+          <FilterComponent key={filter.id} template={template} filter={filter} />
+        ))}
+      </div>
+
+      <ViewAll template={template} />
+    </div>
+  )
+}
+
+const FilterComponent: React.FC<{ template: TemplateDetails; filter: Filter }> = ({
+  template,
+  filter,
+}) => {
+  const templateType = template.code
+  const { loading, applicationCount } = useListApplications({
+    type: templateType,
+    perPage: 1,
+    ...filter.query,
+  })
+
+  const applicationListUserRole =
+    filter.userRole === PermissionPolicyType.Apply ? USER_ROLES.APPLICANT : USER_ROLES.REVIEWER
+
+  const constructLink = () =>
+    `/applications?type=${templateType}&user-role=${applicationListUserRole}&${Object.entries(
+      filter.query
+    )
+      .map(([key, value]) => `${key}=${value}`)
+      .join('&')}`
```

Comment:
Maybe nice to keep this in a separated helper file -- don't. know which one, but there is a folder for `helpers`, it might be nice to have in there helpers related to "updating the URL"

Reply:
There's actually a way you can pass state data directly along with a `.push` if you want to look into that (I used it a little bit in `AuthenticatedWrapper.tsx`)

So you could just keep the parameters bundled in the object, pass them through, then use "updateQuery" in the receiving page (ie. applications) to set the URL etc.

Reply:
> Maybe nice to keep this in a separated helper file -- don't. know which one, but there is a folder for helpers, it might be nice to have in there helpers related to "updating the URL"

I was wondering if we intend to use it anywhere else ? If so it would be good to move to a common place. 

> There's actually a way you can pass state data directly along with a .push if you want to look into that (I used it a little bit in AuthenticatedWrapper.tsx)
> So you could just keep the parameters bundled in the object, pass them through, then use "updateQuery" in the receiving page (ie. applications) to set the URL etc.

We already have quite a few useEffect in `applications` pages, I wonder if this would complicate it further. I really like the idea of just navigating to applications page with just the url and for it to do the job of applying filters to the list

Reply:
> We already have quite a few useEffect in `applications` pages, I wonder if this would complicate it further. I really like the idea of just navigating to applications page with just the url and for it to do the job of applying filters to the list

Yeah that seems fine in this case, since it's just creating new URL from scratch. It's only really important when you want to update existing queries.

Reply:
Well, we can leave to do when needed to use in another place. That's fine. I'm just keen to keep the code cleaner by having functions that does only one thing. We have been doing that in many places which is pretty cool. I though it was a nice one to point out, it is doing a separated logic inside here.